### PR TITLE
Provide node disks API

### PR DESCRIPTION
**Pull request by tcibinan**
Relates to #1096.

The pull request brings support for node disks api. From now node disks history data can be accessed via a new API method `GET /cluster/node/{name}/disks`. All attached disks of all nodes are automatically registered including os, data, swap and autoscaled disks.

Additionally the pull request expands run price calculation capabilities. Two additional fields are added to pipeline run entity that can be used in billing reports:
- `computePricePerHour` which specifies only virtual machine price per hour,
- `diskPricePerHour` which specifies 1GB disk price per hour.

Moreover overall run `pricePerHour` gets updated once compute node is assigned to the run to more precisely reflect node disks cost.

**Notice** that node disks API only covers runs that were created after the patch is applied.

### CODE REVIEW

File: api/src/main/java/com/epam/pipeline/dao/cluster/NodeDiskDao.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor
+public class NodeDiskDao extends NamedParameterJdbcDaoSupport {
+    
+    private final String insertNodeDiskQuery;
+    private final String loadNodeDisksByNodeIdQuery;
+    private final String deleteNodeDisksByNodeIdQuery;
+
+    public List<NodeDisk> insert(final String nodeId, final List<DiskRegistrationRequest> requests) {
```
Comment (mzueva) [inline api/src/main/java/com/epam/pipeline/dao/cluster/NodeDiskDao.java:24]: Please add `Transactional` annotation

File: api/src/test/java/com/epam/pipeline/dao/cluster/NodeDiskDaoTest.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    private static final String NULL_NODE_ID = null;
+    private static final Long SIZE = 1L;
+    private static final Long NULL_SIZE = null;
+
+    @Autowired
+    private NodeDiskDao dao;
+
+    @After
+    public void deleteDisksByNodeId() {
+        dao.deleteByNodeId(NODE_ID);
```
Comment (mzueva) [inline api/src/test/java/com/epam/pipeline/dao/cluster/NodeDiskDaoTest.java]: You can just add `@Transactional` annotation to the class and all DB changes will be rolled-back.

**Review by mzueva — CHANGES_REQUESTED** [review]

File: api/src/main/java/com/epam/pipeline/dao/cluster/NodeDiskDao.java (hunk 1 above)
Comment (tcibinan) [reply api/src/main/java/com/epam/pipeline/dao/cluster/NodeDiskDao.java:24]: Added.

File: api/src/test/java/com/epam/pipeline/dao/cluster/NodeDiskDaoTest.java (hunk 2 above)
Comment (tcibinan) [reply api/src/test/java/com/epam/pipeline/dao/cluster/NodeDiskDaoTest.java]: That's good. Thanks. Added.

**Review by mzueva — APPROVED** [review]

**Pull request merged**